### PR TITLE
Used DATASET resource type instead of REPOSITORY for the dataset entity

### PR DESCRIPTION
**Pull request by anandjakhaniya**
- Used the `ModelDBServiceResourceTypes.DATASET` instead of `ModelDBServiceResourceTypes.REPOSITORY` for the dataset entity
- **Jira Ticket:** https://vertaai.atlassian.net/browse/VR-9563
- **Slack thread:** https://vertaai.slack.com/archives/C01JVHUHKR6/p1610683361002700

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 1]
```diff
@@ -216,13 +216,19 @@ public RepositoryDAORdbImpl(
   private void checkIfEntityAlreadyExists(
       Session session, Workspace workspace, String name, RepositoryTypeEnum repositoryType) {
     List<Long> repositoryEntityIds = getRepositoryEntityIdsByName(session, name, repositoryType);
+    ModelDBServiceResourceTypes modelDBServiceResourceTypes =
+        ModelDBServiceResourceTypes.REPOSITORY;
+    if (repositoryType.equals(RepositoryTypeEnum.DATASET)) {
+      modelDBServiceResourceTypes = ModelDBServiceResourceTypes.DATASET;
+    }
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:244]: This has been repeated a lot in the code. Can we have this a utility function somewhere?

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java [hunk 2]
```diff
@@ -145,32 +147,55 @@ private void cleanRepositories(Session session) {
...
-        "Repository cleaned successfully : Cleaned repositories count {}", repositoryIds.size());
+        "Repository cleaned successfully : Cleaned repositories count {}",
+        repositoryEntityMap.size());
+  }
+
+  private void cleanUpEntityBasedOnResourceType(
+      Session session,
+      List<RepositoryEntity> repositoryEntities,
+      Map<String, RepositoryEntity> repositoryEntityMap,
+      ModelDBResourceEnum.ModelDBServiceResourceTypes repository) {
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java]:
```suggestion
      ModelDBResourceEnum.ModelDBServiceResourceTypes resourceType) {
```
For clarity

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java [hunk 3]
```diff
@@ -145,32 +147,55 @@ private void cleanRepositories(Session session) {
...
+          repositoryEntities,
+          repositoryEntityMap,
+          ModelDBResourceEnum.ModelDBServiceResourceTypes.DATASET);
     }
 
     LOGGER.debug(
-        "Repository cleaned successfully : Cleaned repositories count {}", repositoryIds.size());
+        "Repository cleaned successfully : Cleaned repositories count {}",
+        repositoryEntityMap.size());
+  }
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java]: Also log number of datasets?

File: backend/src/main/java/ai/verta/modeldb/versioning/FindRepositoriesQuery.java [hunk 4]
```diff
@@ -282,8 +282,16 @@ private void setOwnerPredicate(
...
+        Set<String> datasetIdSet =
+            RdbmsUtils.getResourceIdsFromUserWorkspaces(
+                authService,
+                roleService,
+                ModelDBResourceEnum.ModelDBServiceResourceTypes.DATASET,
+                userInfoList);
         List<Long> resourcesIds =
             repositoryIdSet.stream().map(Long::parseLong).collect(Collectors.toList());
+        resourcesIds.addAll(
+            datasetIdSet.stream().map(Long::parseLong).collect(Collectors.toList()));
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/FindRepositoriesQuery.java]: I'm confused why we're adding datasets in FindRepository.

**Review by conradoverta — COMMENTED** [review]
A few minor suggestions but one big item on the FindRepositories.

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:244]: Extracted in the ModelDBUtils

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java [hunk 5]
```diff
@@ -145,32 +147,55 @@ private void cleanRepositories(Session session) {
...
+      cleanUpEntityBasedOnResourceType(
+          session,
+          repositoryEntities,
+          repositoryEntityMap,
+          ModelDBResourceEnum.ModelDBServiceResourceTypes.REPOSITORY);
+
+      cleanUpEntityBasedOnResourceType(
+          session,
+          repositoryEntities,
+          repositoryEntityMap,
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java]:
```suggestion
          datasetEntityMap,
```

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java]: renamed

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java]: Added log for dataset as well

File: backend/src/main/java/ai/verta/modeldb/versioning/FindRepositoriesQuery.java (hunk 4 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/FindRepositoriesQuery.java]: it was irrelevant there so removed

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java (hunk 5 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 6]
```diff
@@ -351,13 +357,18 @@ public RepositoryEntity getProtectedRepositoryById(
   private Optional<RepositoryEntity> getRepositoryByName(
       Session session, String name, Workspace workspace, RepositoryTypeEnum repositoryType) {
     List<Long> repositoryIds = getRepositoryEntityIdsByName(session, name, repositoryType);
+    ModelDBServiceResourceTypes modelDBServiceResourceTypes =
+        ModelDBServiceResourceTypes.REPOSITORY;
+    if (repositoryType.equals(RepositoryTypeEnum.DATASET)) {
+      modelDBServiceResourceTypes = ModelDBServiceResourceTypes.DATASET;
+    }
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:386]: We should replace this with the helper function, but not sure how to do it right now.
```suggestion
    // TODO: replace with the helper function
    ModelDBServiceResourceTypes modelDBServiceResourceTypes =
        ModelDBServiceResourceTypes.REPOSITORY;
    if (repositoryType.equals(RepositoryTypeEnum.DATASET)) {
      modelDBServiceResourceTypes = ModelDBServiceResourceTypes.DATASET;
    }
```

**Review by conradoverta — APPROVED** [review]
Added a TODO but LGTM otherwise

**Pull request merged**